src/editor.rs

```diff
@@ -476,6 +479,14 @@ impl<'a, W: TermWrite + Write> Editor<'a, W> {
         self.print_current_buffer(false)
     }
 
+    /// Deletes every character preceding the cursor until the beggining of the line
+    pub fn delete_all_before_cursor(&mut self) -> io::Result<()> {
+	    while self.cursor > 0 {
+			self.delete_before_cursor();
```

Comment:
Every call to `delete_before_cursor` will reprint the current buffer. It would be better to call `cur_buf_mut!(self).remove(0, self.cursor)`. Other than that, nice.